mobile/native/src/dlc/node.rs

```diff
@@ -370,7 +377,11 @@ impl Node {
                     "Finished rollover protocol"
                 );
 
-                position::handler::set_position_state(PositionState::Open)?;
+                let position = update_position_after_rollover()
```

Comment:
🔧 these two updates should probably run within the same database transaction.

Reply:
I totally agree, but this requires reworking `get_positions`, which is used in 11 spots.